### ISSUE TITLE
google_secret_manager_secret: bad compute of project_id when importing resources

### ISSUE THREAD

**Issue by Kaawin1** [hashicorp/terraform-provider-google#13734]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to the `modular-magician` user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If an issue is assigned to a user, that user is claiming responsibility for the issue. If an issue is assigned to `hashibot`, a community member has claimed the issue already.

### Terraform Version

Terraform v1.2.4
terragrunt version v0.38.9
Provider version: v4.47.0

### Affected Resource(s)

google_secret_manager_secret
google_secret_manager_secret_version

### Terraform Configuration Files

```tf
provider "google" {
  project = var.project_id
  region  = var.region
}

provider "sops" {}


resource "google_secret_manager_secret" "this" {
  for_each = local.secrets_name

  
  secret_id = each.key

  replication {
    automatic = true
  }
}

resource "google_secret_manager_secret_version" "this" {
  for_each = local.secrets_name

  secret      = google_secret_manager_secret.this[each.key].id
  secret_data = local.secrets[each.key].value
}
```

When importing a secret and a secret version, terraform does not check the project id when not explicitly specify in the resource. So if I import a prod secret in a preprod stack, terraform should detect that the project_id of the secret is not correct.

### Expected Behavior

As specify in the [doc](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/secret_manager_secret#project), if the project_id is not specify, the provider project should be used.

### Actual Behavior

Terraform does not detect that the project_id is not the right one. After an import, the project is not changed if not specified explicitly in the resource.

### Steps to Reproduce

1. Create a secret in two differents GCP projects (project A et B). Let's say `test-secret`
2. In project A terragrunt' stack, run `terragrunt import 'google_secret_manager_secret.this["test-import"]' projects/projectB/secrets/test-secret`. So here the project ID is not good since we are in the project A stack
3. Run `terragrunt import 'google_secret_manager_secret_version.this["test-secret"]' projects/projectB/secrets/test-import/versions/latest`
4. Run `terragrunt plan` 

Terragrunt does not detect any changes even though the project set during the import is not good. I fixed this bug by specifying the project_id explicitly in the google resource.

```tf
resource "google_secret_manager_secret" "this" {
  for_each = local.secrets_name

  project_id = var.project_id
  secret_id = each.key

  replication {
    automatic = true
  }
}
```
But it should not be necessary

```tf
# google_secret_manager_secret.this["test-secret"]:
resource "google_secret_manager_secret" "this" {
    id          = "projects/projectB/secrets/test-secret"
    labels      = {}
    name        = "projects/idProjectB/secrets/test-secret"
    project     = "projectB"
    secret_id   = "test-secret"
}

# google_secret_manager_secret_version.this["test-secret"]:
resource "google_secret_manager_secret_version" "this" {
    id           = "projects/idProjectB/secrets/test-secret/versions/2"
    name         = "projects/idProjectB/secrets/test-secret/versions/2"
    secret       = "projects/projectB/secrets/test-secret"
    secret_data  = (sensitive value)
}
```

This is the output of `terragrunt show` command. The state shows the bad project because it follows the import correct (which is incorrect) but a new plan do not suggest to fix the value until I explicitly specify the project_id in the resource.

**Comment by edwardmedia** [comment]
same issue of https://github.com/hashicorp/terraform-provider-google/issues/13449

**Comment by melinath** [comment]
I think this is separate from https://github.com/hashicorp/terraform-provider-google/issues/13449. This sounds like the expected behavior - the project field is Optional + Computed, so it will silently preserve the value set on import rather than overwriting it (and recreating the resource). Fixing this would require overhauling how we treat project fields in the provider as a whole.

**Comment by SarahFrench** [comment]
👋 @NickElliot - would your work in 5.0.0 resolve this issue?

**Comment by melinath** [comment]
@NickElliot ping re: above

**Comment by NickElliot** [comment]
Sorry, slipped my mind after I started looking into it!

The update I implemented will not resolve the base import problem -- it being a customize diff, which would never be called during the import process. But it would attempt to revise the plan once that was called, because the logic of the customize diff checks if there is a configured value in the state (calling a diff.Get() specifically checks for "new" values), and if not present follows the standard "default" value order.

**Comment by rileykarson** [comment]
Looking over this again, this should resolve the issue- since the project id value is no longer sticky due to the CustomizeDiff, the incorrectly specified project should get detected.

> So it does resolve the need for the configuration file to be further updated to adjust the state (but would require a new Apply), it would not resolve what appears to be the main issue of this ticket.

This won't fix it at import time- the user specified the wrong value, so we don't expect it to- and this will be resolved by Terraform detecting drift between the provider default project and the resource's inferred project.

Closing as resolved in https://github.com/hashicorp/terraform-provider-google/issues/11339. If this isn't fixed by `5.0.0` next week, please feel free to reopen @Kaawin1!
